asserts/findwildcard_test.go

```diff
@@ -129,11 +129,151 @@ func (fs *findWildcardSuite) TestFindWildcardSomeErrors(c *check.C) {
 	c.Check(err, check.ErrorMatches, "expected a regular file: .*")
 }
+
+func (fs *findWildcardSuite) TestFindWildcardSequence(c *check.C) {
+	top := filepath.Join(c.MkDir(), "top")
+
+	err := os.MkdirAll(top, os.ModePerm)
+	c.Assert(err, check.IsNil)
+
+	files := []string{
+		"s1/3/active.1",
+		"s1/3/active.2",
+		"s1/2/active",
+		"s1/2/active.1",
+		"s1/1/active",
+	}
+	for _, fn := range files {
+		err := os.MkdirAll(filepath.Dir(filepath.Join(top, fn)), os.ModePerm)
+		c.Assert(err, check.IsNil)
+		err = ioutil.WriteFile(filepath.Join(top, fn), nil, os.ModePerm)
+		c.Assert(err, check.IsNil)
+	}
+
+	var res [][]string
+	foundCb := func(relpath []string) error {
+		res = append(res, relpath)
+		return nil
+	}
+
+	sort := func() {
+		for _, r := range res {
+			sort.Strings(r)
+		}
+	}
+
+	// ascending
+
+	err = findWildcard(top, []string{"s1", "#>", "active*"}, 1, foundCb)
+	c.Assert(err, check.IsNil)
+	sort()
+	c.Check(res, check.DeepEquals, [][]string{
+		{"s1/2/active", "s1/2/active.1"},
+		{"s1/3/active.1", "s1/3/active.2"},
+	})
+
+	res = nil
+	err = findWildcard(top, []string{"s1", "#>", "active*"}, 2, foundCb)
+	c.Assert(err, check.IsNil)
+	sort()
+	c.Check(res, check.DeepEquals, [][]string{
+		{"s1/3/active.1", "s1/3/active.2"},
+	})
+
+	res = nil
+	err = findWildcard(top, []string{"s1", "#>", "active*"}, 3, foundCb)
+	c.Assert(err, check.IsNil)
+	c.Check(res, check.HasLen, 0)
+
+	res = nil
+	err = findWildcard(top, []string{"s1", "#>", "active*"}, -1, foundCb)
+	c.Assert(err, check.IsNil)
+	sort()
+	c.Check(res, check.DeepEquals, [][]string{
+		{"s1/1/active"},
+		{"s1/2/active", "s1/2/active.1"},
+		{"s1/3/active.1", "s1/3/active.2"},
+	})
+
+	// descending
+
+	res = nil
+	err = findWildcard(top, []string{"s1", "#<", "active*"}, -1, foundCb)
+	c.Assert(err, check.IsNil)
+	sort()
+	c.Check(res, check.DeepEquals, [][]string{
+		{"s1/3/active.1", "s1/3/active.2"},
+		{"s1/2/active", "s1/2/active.1"},
+		{"s1/1/active"},
+	})
+
+	res = nil
+	err = findWildcard(top, []string{"s1", "#<", "active*"}, 3, foundCb)
+	c.Assert(err, check.IsNil)
+	sort()
+	c.Check(res, check.DeepEquals, [][]string{
+		{"s1/2/active", "s1/2/active.1"},
+		{"s1/1/active"},
+	})
+
+	res = nil
+	err = findWildcard(top, []string{"s1", "#<", "active*"}, 2, foundCb)
+	c.Assert(err, check.IsNil)
+	sort()
+	c.Check(res, check.DeepEquals, [][]string{
+		{"s1/1/active"},
+	})
+
+	res = nil
+	err = findWildcard(top, []string{"s1", "#<", "active*"}, 1, foundCb)
+	c.Assert(err, check.IsNil)
+	c.Check(res, check.HasLen, 0)
+
+	// missing dir
+	res = nil
+	err = findWildcard(top, []string{"s2", "#<", "active*"}, 1, foundCb)
+	c.Assert(err, check.IsNil)
+	c.Check(res, check.HasLen, 0)
+}
+
+func (fs *findWildcardSuite) TestFindWildcardSequenceSomeErrors(c *check.C) {
+	top := filepath.Join(c.MkDir(), "top-errors")
+
+	err := os.MkdirAll(top, os.ModePerm)
+	c.Assert(err, check.IsNil)
+
+	files := []string{
+		"s1/1/active",
+		"s2/a/active.1",
+		"s3/-9/active.1",
+	}
```

Comment:
For completeness, can you add `"s4/01/active"`? AFAICT we'll get nil error in this case.